lib/fluent/plugin/base.rb

```diff
@@ -53,14 +53,12 @@ def fluentd_worker_id
       end
 
       def configure(conf)
-        if Fluent::Engine.supervisor_mode || (conf.respond_to?(:for_this_worker?) && conf.for_this_worker?)
-          workers = if conf.target_worker_ids && !conf.target_worker_ids.empty?
-                      conf.target_worker_ids.size
-                    else
-                      1
-                    end
-          system_config_override(workers: workers)
+        raise ArgumentError, "BUG: type of conf must be Fluent::Config::Element, but #{conf.class} is passed." unless conf.is_a?(Fluent::Config::Element)
+
+        if (Fluent::Engine.supervisor_mode && !conf.target_worker_ids.empty?) || conf.for_this_worker?
```

Comment:
```suggestion
        if conf.for_this_worker? || (Fluent::Engine.supervisor_mode && !conf.for_every_workers?)
```

How about this?

* It is easier to understand if the `conf.for_this_worker?` condition is first.
* We can use `for_every_workers?` instead of `target_worker_ids.empty?`. It is easier to understand the intention.

Reply:
I changed it because it is easier to understand.